apps/native/app/(tabs)/wallet/[walletId].tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useMemo, useState } from "react";
+import {
+    SafeAreaView,
+    ScrollView,
+    StyleSheet,
+    Text,
+    TouchableOpacity,
+    View,
+} from "react-native";
+import { Stack, useLocalSearchParams } from "expo-router";
+import { WalletActivity } from "../../../components/wallet/WalletActivity";
+import { WalletDetails } from "../../../components/wallet/WalletDetails";
+import { Colors } from "../../../constants/Colors";
+import { getPlaceholderWalletActivity } from "../../../data/walletActivityPlaceholder";
+
+const mapParamsToString = (param: string | string[] | undefined) =>
+    Array.isArray(param) ? param.join("") : param ?? "";
+
+export default function WalletDetail() {
+    const params = useLocalSearchParams();
+    const walletId = mapParamsToString(params.walletId);
+    const walletName = mapParamsToString(params.name) || "Wallet";
+    const balanceValue = Number.parseFloat(mapParamsToString(params.balance));
+    const walletBalance = Number.isFinite(balanceValue) ? balanceValue : 0;
+    const [activeTab, setActiveTab] = useState<"activity" | "details">(
+        "activity",
+    );
+
+    const { pending, completed, completedLabel } = useMemo(
```

Comment:
If the computation is cheap and deterministic, useMemo is overkill and should be removed.